Foody/Onboarding/Register/RegisterViewController.swift

```diff
@@ -6,24 +6,161 @@
+    @IBOutlet weak var signUpClicked: UIButton!
+    
+    let datePicker = UIDatePicker()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        // Do any additional setup after loading the view.
+        
+        signUpClicked.layer.cornerRadius = 20.0
+        
+        usernameText.placeholder = "Username"
+        emailText.placeholder = "Email"
+        passwordText.placeholder = "Password"
+        dateText.placeholder = "Date of Birth"
+        
+        datePicker.addTarget(self, action: #selector(dateChange(datePicker:)), for: UIControl.Event.valueChanged)
+        datePicker.frame.size = CGSize(width: 0, height: 300)
+        datePicker.preferredDatePickerStyle = .wheels
+        
+        datePicker.maximumDate = Date()
+        
+        createDatePicker()
     }
     
-
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+    @objc func dateChange(datePicker: UIDatePicker) {
+        dateText.text = formatDate(date: datePicker.date)
+    }
+    
+    func createDatePicker() {
+        
+      
+        let toolbar = UIToolbar()
+        toolbar.sizeToFit()
+        
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target: nil, action: #selector(doneTapped))
+        toolbar.setItems([doneButton], animated: true)
+        dateText.inputAccessoryView = toolbar
+        
+        dateText.inputView = datePicker
+        datePicker.datePickerMode = .date
+    }
+    @objc func doneTapped () {
+        
+        let formatter = DateFormatter()
+        formatter.dateStyle = .medium
+        formatter.timeStyle = .none
+        
+        dateText.text = formatter.string(from: datePicker.date)
+        self.view.endEditing(true)
+    }
+    
+    func formatDate(date: Date) -> String {
+        
+        let formatter = DateFormatter()
+        formatter.dateFormat = "dd/MM/yyyy"
+        return formatter.string(from: date)
+    }
+    
+    func dateDayCount(date:String,format:String = "dd/MM/yyyy") -> Int {
+        
+        let dateFormatter = DateFormatter()
+        
+        dateFormatter.locale = Locale.current
+        dateFormatter.timeZone = TimeZone.autoupdatingCurrent
+        dateFormatter.timeStyle = .short
+        dateFormatter.dateFormat = format
+        
+        let calender = Calendar.current
+        let firstDate = calender.startOfDay(for: dateFormatter.date(from: date)!)
+        let twiceDate = calender.startOfDay(for: Date())
+        let companents = calender.dateComponents([.year], from: firstDate, to: twiceDate)
+        
+        if companents.year! >= 15 {
+            
+            makeAlert(title: "Registration Successful.", message:"You are welcome. :)" )
```

Comment:
This is not right place to say this message. It needs to be after firebase sign up.